core/src/main/java/org/elasticsearch/index/query/AbstractQueryBuilder.java

```diff
@@ -195,7 +195,7 @@ protected boolean doEquals(QB other) {
 
     @Override
     public final int hashCode() {
-        return 31 * Objects.hash(getClass(), queryName, boost) + doHashCode();
+        return Objects.hash(getClass(), queryName, boost, doHashCode());
```

Comment:
This is related to https://github.com/elastic/elasticsearch/issues/12145, right?